optaplanner-persistence/optaplanner-persistence-jaxb/src/test/java/org/optaplanner/persistence/jaxb/api/score/AbstractScoreJaxbXmlAdapterTest.java

```diff
@@ -103,22 +106,25 @@
             throw new IllegalStateException("Marshalling or unmarshalling for input (" + input + ") failed.", e);
         }
         assertEquals(expectedScore, output.getScore());
+
+        String newLine = System.lineSeparator();
         String regex;
         if (expectedScore != null) {
-            regex = "\\{\n" // Opening bracket
```

Comment:
Proposal B)
How about we use "\r?\n" instead of "\n"?

Proposal C) Why do we assert the formatting of the text? Is that needed?
Normally "\\s*" includes all whitespace, including \r and \n.
Does this regex work for those 2 lines?

```
regex = "\\{\\s*\"([\\w]+)\"\\s:\\s\\{"
 + ...
```

Reply:
If C) works, can we go with that? I don't think we need to test for formatting.

"\s*" (which is escaped written as "\\s*"), means 0 or more whitespace characters.

Reply:
Or Proposal D): The official Java regex for newline since Java 8: "\R" (https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html#lineending)

Reply:
Wouldn't using jackson ObjectMapper (or similar) make sense here? We don't really care about how the JSON looks like, just that the JSON has certain fields/properties (which we can check via ObjectMapper:
```
JsonNode parent= new ObjectMapper().readTree(json);
String score = parent.get("score").asText();
```

Reply:
+1 to using `\R` for Java regex. What would we use for new line characters in the error messages?
-1 to using `ObjectMapper`, I don't think we should introduce Jackson dependencies since this code deals with the JAXB JSON standard. We do use `ObjectMapper` in the `optaplanner-persistence-jackson` module. If we do decide to use it, it's outside the scope of this PR.

Reply:
String.format(...) is a thing, and use "%n" (https://dzone.com/articles/java-string-format-examples) (note it varargs are not type safe, it basically `sprintf`).

How about using `JAXBContext` to create a new unmarshaller to read the json?

Reply:
`%n` looks reasonable. Let's see if we can get consensus on which newline char to use. As for the rest, a bit of scope creep here, the purpose of this PR was to fix the broken Windows tests. We can address how we assert the Json output in a separate PR.

Reply:
The JSON format is whitespace (and therefore newline) agnostic. Whether one puts newlines and 4 space indents into a JSON file is a matter of taste (= code style). 
We need to test if it's formatted as JSON.
Do we need to test if the JSON "code style" is adhered too? I am not sure. If we don't, then we can just put `\\s*`, this will match all and any whitespacing (including linefeed and newline characters). If we do want to test it, then this is fine.
Either way, +1 to merge.

Reply:
Hmm, good points. I don't see why we shouldn't test for JSON code style. If anything it's a more strict test, so I am in favour of keeping it. Are there any examples where testing for "style" rather than just "format" is not a good choice?